tests/conftest.py

```diff
@@ -3143,12 +3156,18 @@ def factory(
         log.info(f"Waiting for {waiting_time}sec to generate msg")
         time.sleep(waiting_time)
 
-        # Check messages are sent and received
-        threads = amq.run_in_bg(
-            namespace=kafka_namespace, value=value, since_time=since_time
-        )
+        if validate_messages:
+            if run_in_bg:
+                # Check messages are sent and received
+                threads = amq.run_in_bg(
+                    namespace=kafka_namespace, value=value, since_time=since_time
+                )
+                return amq, threads
+            else:
+                amq.validate_messages_are_produced()
+                amq.validate_messages_are_consumed()
 
-        return amq, threads
+        return amq
 
```

Comment:
you may as well have to check all the tests for the respective workloads and make necessary changes

Reply:
Not needed as the newly introduced flags are by default set to True. I have plans to trigger the entire suite if I find any regressions will make the appropriate changes